### DS102-L7-Hands-on-Helper
### Respectfully please do not copy and paste from this notebook. Consequence will be  to receive a lovely 0 :) We really want these notebooks to be a help to you to understand the concepts but for you to do your own work:)

In [1]:
### This notebook is meant to help you understand
### the Lesson 7 hands on!! 

In [2]:
### Lesson 7 uses a built in dataset which means the
### data is already for you and you don't have to load
### it in!

In [3]:
### We will use a build in dataset on plants!!!

<img srcset="https://cdn.pixabay.com/photo/2017/11/28/10/23/nature-2983254_960_720.jpg 1x, https://cdn.pixabay.com/photo/2017/11/28/10/23/nature-2983254_1280.jpg 2x" src="https://cdn.pixabay.com/photo/2017/11/28/10/23/nature-2983254_960_720.jpg" alt="Free photos of Nature" width="960" height="720.0000000000018" style="height: auto">

So the PlantGrowth dataset has 3 groups in it, the control group, treatment1, and treatment2. Here we will be comparing treatment1 to treatment2.

First let's run a code in R to view our built-in dataset of PlantGrowth:

In [4]:
View(PlantGrowth)

weight group
1  4.17   ctrl 
2  5.58   ctrl 
3  5.18   ctrl 
4  6.11   ctrl 
5  4.50   ctrl 
6  4.61   ctrl 
7  5.17   ctrl 
8  4.53   ctrl 
9  5.33   ctrl 
10 5.14   ctrl 
11 4.81   trt1 
12 4.17   trt1 
13 4.41   trt1 
14 3.59   trt1 
15 5.87   trt1 
16 3.83   trt1 
17 6.03   trt1 
18 4.89   trt1 
19 4.32   trt1 
20 4.69   trt1 
21 6.31   trt2 
22 5.12   trt2 
23 5.54   trt2 
24 5.50   trt2 
25 5.37   trt2 
26 5.29   trt2 
27 4.92   trt2 
28 6.15   trt2 
29 5.80   trt2 
30 5.26   trt2

Above we can see our data. 

### What I am showing you below is not necessary for your hands on, but will explain more of what we are doing and be nice to be exposed to. However, if you like to skip this section you can, Ill let you know when you want to pick it up again:)

To view our number of columns/rows:

In [18]:
#number of columns uses ncol() function and in the () you place the dataset name
ncol(PlantGrowth)

[1] 2

In [19]:
#number of rows uses nrow() function and in the () you place the dataset name
nrow(PlantGrowth)

[1] 30

In [16]:
#I don't like to see the pink messages when I load libraries so I use this code:
suppressPackageStartupMessages(library(dplyr))

In [20]:
#The reason I am running this code below is because I know we want to do a dependent t test
#To do this test we need to have the groups equal in number so by running the count of each
#group I can see that our groups are even, see below how we have 10 of each? 10 of the ctrl,
#10 of trt1, and 10 of trt2. Now we know we can create our vectors and they will be of the
#same length! woo hoo!
PlantGrowth %>% count(group)

group n 
1 ctrl  10
2 trt1  10
3 trt2  10

### Ok start back here if you skipped the last section!

We are going to create two vectors just like in your hands on:

In [23]:
Trt1 <- PlantGrowth[11:20,]
Trt2 <- PlantGrowth[21:30,]

In [29]:
#The reason I am viewing this is to show you this is slightly 
#different than your hands on data! Do you see how it has 
#another column there? So we will do something slightly 
#different below in our t test
View(Trt1)

weight group
11 4.81   trt1 
12 4.17   trt1 
13 4.41   trt1 
14 3.59   trt1 
15 5.87   trt1 
16 3.83   trt1 
17 6.03   trt1 
18 4.89   trt1 
19 4.32   trt1 
20 4.69   trt1

In your hands on you will use the vectors they give you. This is extremely similar!

### Now we have our vectors we can do the appropriate t test!

If you have not already, read your hands on and try to guess what test you are going to do! Then come back and continue here:)

So we will be comparing two related groups of equal length! hint! Dependent t test!

Before we dive into the test, let's review the null and alternative hypothesis!

Null is essentially stating that two groups of comparison are the same. The alternative hypothesis is saying that the two groups in question are indeed not the same, but rather significantly different!
(Don't copy and paste that ! :))

Then we will run our test for our p value which we will use to determine if we should reject the null or accept the null. If p < .05 reject the null, if p > .05 accept the null!

Here I am doing the dependent t test from page 4 in your LMS

In [28]:
# first you name it, ours is called the same as the lesson to 
# avoid confusion. Then we use the t.test() function
# next we need to place our vectors! 
# paired= True is because this is a dependent test

t_dep <- t.test(Trt1$weight, Trt2$weight, paired = TRUE)
t_dep


	Paired t-test

data:  Trt1$weight and Trt2$weight
t = -2.8464, df = 9, p-value = 0.0192
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.5524646 -0.1775354
sample estimates:
mean of the differences 
                 -0.865 


### Hold on one minute!

Ok so you can see above the t test is like the example in the lesson on page 4. 
### However:

### Really IMPORTANT

In your hands on, the dataset doesn't have extra columns it only has the one numeric column so when the vector is created, that is all that is needed to represent the data. 
That being said, FOR YOUR HANDS ON YOU DO NOT NEED TO USE THE $ with a column after because you don't have any other 
columns there and the vectors are already separating the data into the groups you will want to use! So you will use name of vector one, name of vector 2 Ill illustrate below:

In [30]:
# t_dep <- t.test(VECTOR1, VECTOR2, paired = TRUE)
# t_dep

So above you will use that format and place the name of the
first vector in the spot of VECTOR1 and the name of the second
vector in place of VECTOR2 then run it ! And you will see your p value.

Ok NOW going back up to view our results here we see that our p value was  0.0192, so is that significant or not? Would you reject the null or accept it?

pvalue < .05 so we will reject the null!

In [31]:
#Hope this example was helpful to you!! <3 Mia